packages/world/src/modules/index/IndexModule.sol

```diff
@@ -0,0 +1,63 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import { console } from "forge-std/console.sol";
+
+import { StoreSwitch } from "@latticexyz/store/src/StoreSwitch.sol";
+
+import { IWorld } from "../../interfaces/IWorld.sol";
+import { IModule } from "../../interfaces/IModule.sol";
+
+import { ROOT_NAMESPACE, CORE_MODULE_NAME } from "../../constants.sol";
+import { WorldContext } from "../../WorldContext.sol";
+import { ResourceSelector } from "../../ResourceSelector.sol";
+
+import { ReverseMappingHook } from "./ReverseMappingHook.sol";
+import { ReverseMapping } from "./tables/ReverseMapping.sol";
+
+/**
+ * This module deploys a hook that is called when a value is set in the `sourceTableId`
+ * provided in the install methods arguments. The hook keeps track of a "reverse mapping"
+ * from value to list of keys with this value. This mapping is stored in a table registered
+ * by the module at the `targetTableId` provided in the install methods arguments.
+ *
+ * Note: for now this module only supports tables with single keys, no composite keys.
+ * Support for composite keys can be added by using a parallel array to store the key in the target table.
+ *
+ * Note: this module currently expects to be `delegatecalled` via World.installRootModule.
+ * Support for installing it via `World.installModule` depends on `World.callFrom` being implemented.
+ */
+contract IndexModule is IModule, WorldContext {
```

Comment:
nit: `IndexModule` doesn't feel self explanatory to me, maybe `ReverseMappingModule`? or `IndexedDataModule`?